packages/mds-api-authorizer/index.ts

```diff
@@ -15,30 +15,38 @@ const {
 } = process.env
 
 export type ApiAuthorizer = (req: express.Request) => ApiAuthorizerClaims | null
+export type WsAuthorizer = (req: string) => ApiAuthorizerClaims | null
+
+const decoders: { [scheme: string]: (token: string) => ApiAuthorizerClaims } = {
+  bearer: (token: string) => {
+    const {
+      sub: principalId,
+      scope,
+      [TOKEN_PROVIDER_ID_CLAIM]: provider_id = null,
+      [TOKEN_USER_EMAIL_CLAIM]: user_email = null,
+      ...claims
+    } = decode(token)
+    return { principalId, scope, provider_id, user_email, ...claims }
+  },
+  basic: (token: string) => {
+    const [principalId, scope] = Buffer.from(token, 'base64')
+      .toString()
+      .split('|')
+    return { principalId, scope, provider_id: principalId, user_email: null }
+  }
+}
 
 export const AuthorizationHeaderApiAuthorizer: ApiAuthorizer = req => {
   if (req.headers && req.headers.authorization) {
     const [scheme, token] = req.headers.authorization.split(' ')
-    const decoders: { [scheme: string]: () => ApiAuthorizerClaims } = {
-      bearer: () => {
-        const {
-          sub: principalId,
-          scope,
-          [TOKEN_PROVIDER_ID_CLAIM]: provider_id = null,
-          [TOKEN_USER_EMAIL_CLAIM]: user_email = null,
-          ...claims
-        } = decode(token)
-        return { principalId, scope, provider_id, user_email, ...claims }
-      },
-      basic: () => {
-        const [principalId, scope] = Buffer.from(token, 'base64')
-          .toString()
-          .split('|')
-        return { principalId, scope, provider_id: principalId, user_email: null }
-      }
-    }
     const decoder = decoders[scheme.toLowerCase()]
-    return decoder ? decoder() : null
+    return decoder ? decoder(token) : null
   }
   return null
 }
+
+export const WebSocketAuthorizer: WsAuthorizer = jwt => {
+  const [scheme, token] = jwt.split(' ')
```

Comment:
Assuming this also supports basic auth, the _token_ isn't technically a JWT. Also, technically, this argument could be `header` and then the authorizer that accepts the `req` could call into this one.